Sources/Bow/Typeclasses/Decidable.swift

```diff
@@ -0,0 +1,247 @@
+        _ f: (Z) -> Either<A, Either<B, Either<C, Either<D, Either<E, FF>>>>>
+    ) -> Kind<Self, Z> {
+        choose(
+            fa,
+            fb,
+            fc,
+            fd,
+            choose(fe, ff, id),
+            f
+        )
+    }
+    
+    static func choose<A, B, C, D, E, FF, G, Z>(
+        _ fa: Kind<Self, A>,
+        _ fb: Kind<Self, B>,
+        _ fc: Kind<Self, C>,
+        _ fd: Kind<Self, D>,
+        _ fe: Kind<Self, E>,
+        _ ff: Kind<Self, FF>,
+        _ fg: Kind<Self, G>,
+        _ f: (Z) -> Either<A, Either<B, Either<C, Either<D, Either<E, Either<FF, G>>>>>>
+    ) -> Kind<Self, Z> {
+        choose(
+            fa,
+            fb,
+            fc,
+            fd,
+            fe,
+            choose(ff, fg, id),
+            f
+        )
+    }
+    
+    static func choose<A, B, C, D, E, FF, G, H, Z>(
+        _ fa: Kind<Self, A>,
+        _ fb: Kind<Self, B>,
+        _ fc: Kind<Self, C>,
+        _ fd: Kind<Self, D>,
+        _ fe: Kind<Self, E>,
+        _ ff: Kind<Self, FF>,
+        _ fg: Kind<Self, G>,
+        _ fh: Kind<Self, H>,
+        _ f: (Z) -> Either<A, Either<B, Either<C, Either<D, Either<E, Either<FF, Either<G, H>>>>>>>
+    ) -> Kind<Self, Z> {
+        choose(
+            fa,
+            fb,
+            fc,
+            fd,
+            fe,
+            ff,
+            choose(fg, fh, id),
+            f
+        )
+    }
+    
+    static func choose<A, B, C, D, E, FF, G, H, I, Z>(
+        _ fa: Kind<Self, A>,
+        _ fb: Kind<Self, B>,
+        _ fc: Kind<Self, C>,
+        _ fd: Kind<Self, D>,
+        _ fe: Kind<Self, E>,
+        _ ff: Kind<Self, FF>,
+        _ fg: Kind<Self, G>,
+        _ fh: Kind<Self, H>,
+        _ fi: Kind<Self, I>,
+        _ f: (Z) -> Either<A, Either<B, Either<C, Either<D, Either<E, Either<FF, Either<G, Either<H, I>>>>>>>>
+    ) -> Kind<Self, Z> {
+        choose(
+            fa,
+            fb,
+            fc,
+            fd,
+            fe,
+            ff,
+            fg,
+            choose(fh, fi, id),
+            f
+        )
+    }
+    
+    static func choose<A, B, C, D, E, FF, G, H, I, J, Z>(
+        _ fa: Kind<Self, A>,
+        _ fb: Kind<Self, B>,
+        _ fc: Kind<Self, C>,
+        _ fd: Kind<Self, D>,
+        _ fe: Kind<Self, E>,
+        _ ff: Kind<Self, FF>,
+        _ fg: Kind<Self, G>,
+        _ fh: Kind<Self, H>,
+        _ fi: Kind<Self, I>,
+        _ fj: Kind<Self, J>,
+        _ f: (Z) -> Either<A, Either<B, Either<C, Either<D, Either<E, Either<FF, Either<G, Either<H, Either<I, J>>>>>>>>>
+    ) -> Kind<Self, Z> {
+        choose(
+            fa,
+            fb,
+            fc,
+            fd,
+            fe,
+            ff,
+            fg,
+            fh,
+            choose(fi, fj, id),
+            f
+        )
+    }
+}
+
+// MARK: - Syntax for Decidable
+public extension Kind where F: Decidable {
+    func choose<B, Z>(_ fb: Kind<F, B>, _ f: (Z) -> Either<A, B>) -> Kind<F, Z> {
```

Comment:
These functions should be static as we have done in [Applicative](https://github.com/bow-swift/bow/blob/master/Sources/Bow/Typeclasses/Applicative.swift#L583) and [Concurrent](https://github.com/bow-swift/bow/blob/master/Sources/BowEffects/Typeclasses/Concurrent.swift#L40) to have an easier way of using them. Also, the return type should be `Kind<F, A>` (as done in the 2 type classes stated above).